analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/TokenTypeAnalyzerTest.Classifier.cs

```diff
@@ -222,4 +222,677 @@ public class Test {
+        ClassifierTestHarness.AssertTokenTypes(
+            $$"""
+              using System;
+              using System.Collections.Generic;
+              public class Test
+              {
+                  public void M(Exception ex)
+                  {
+                      var x = {{expression}};
+                  }
+              }
+              """/*, allowSemanticModel */);
+
+    [DataTestMethod]
+    [DataRow("([k:string], [t:Exception]) => true,", true)]
+    [DataRow("""([s:""], [k:null]) => true,""", true)]
+    [DataRow("([t:String] a, [t:Exception] b) => 1,", true)]
+    [DataRow("([u:System].[t:String] a, [u:System].[t:Exception] b) => 1,", true)]
+    [DataRow("([t:HashSet]<[t:Int32]> a, null) => 1,", true)]
+    [DataRow("([t:List]<[t:List]<[t:Int32]>> a, null) => 1,", true)]
+    [DataRow("([t:Test].[t:Inner], null) => 1,", true)]
+    [DataRow("([t:Inner], null) => 1,", true)]
+    [DataRow("([u:first]: [t:Inner], [u:second]: null) => 1,", true)]
+    [DataRow("""("", [t:ArgumentException] { HResult: > 2 }) => true,""", false)]
+    [DataRow("(([t:Int32], [u:System].[t:String]), second: null) => 1,", false)]
+    public void IdentifierToken_Tuples(string switchBranch, bool allowSemanticModel = true) =>
+        ClassifierTestHarness.AssertTokenTypes(
+            $$"""
+              using System;
+              using System.Collections.Generic;
+              public class Test
+              {
+                  public void M(Object o, Exception exception)
+                  {
+                      var x = (first: o, second: exception) switch
+                      {
+                          {{switchBranch}}
+                          _ => default,
+                      };
+                  }
+                  public class Inner { }
+              }
+              """/*, allowSemanticModel */);
+
+    [DataTestMethod]
+    [DataRow("[t:Exception] ex;", false)]
+    [DataRow("[u:System].Exception ex;", true)]
+    [DataRow("[t:List]<[t:Exception]> ex;", false)]
+    [DataRow("List<[u:System].Exception> ex;", true)]
+    public void IdentifierToken_LocalDeclaration(string declaration, bool allowSemanticModel = true) =>
+        ClassifierTestHarness.AssertTokenTypes(
+            $$"""
+              using System;
+              using System.Collections.Generic;
+              public class Test
+              {
+                  public void M()
+                  {
+                      {{declaration}}
+                  }
+              }
+              """ /*, allowSemanticModel */);
+
+    [DataTestMethod]
+    [DataRow("__refvalue([u:tf], Exception)", false)]
+    [DataRow("__refvalue(tf, [t:Exception])", false)]
+    [DataRow("__refvalue(tf, [u:System].Exception)", true)]
+    [DataRow("__refvalue(tf, System.[t:Exception])", false)]
+    public void IdentifierToken_Type_RefValue(string refValueExpression, bool allowSemanticModel = true) =>
```

Comment:
See https://www.codeproject.com/Articles/38695/UnCommon-C-keywords-A-Look for details about `__refvalue`